frontend/src/employee-frontend/components/messages/UndoMessageNotification.tsx

```diff
@@ -0,0 +1,103 @@
+// SPDX-FileCopyrightText: 2017-2022 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import differenceInSeconds from 'date-fns/differenceInSeconds'
+import React, { useCallback, useContext, useEffect, useState } from 'react'
+import { useTheme } from 'styled-components'
+
+import HelsinkiDateTime from 'lib-common/helsinki-date-time'
+import InlineButton from 'lib-components/atoms/buttons/InlineButton'
+import { FixedSpaceColumn } from 'lib-components/layout/flex-helpers'
+import Toast from 'lib-components/molecules/Toast'
+import { faCheck, faRedo } from 'lib-icons'
+
+import { useTranslation } from '../../state/i18n'
+
+import { CancelableMessage, MessageContext } from './MessageContext'
+import { undoMessage } from './api'
+
+export default React.memo(function UndoMessageNotification() {
+  const { cancelableMessage } = useContext(MessageContext)
+  return cancelableMessage ? (
+    <UndoMessageToast cancelableMessage={cancelableMessage} />
+  ) : null
+})
+
+const UndoMessageToast = React.memo(function UndoMessageToast({
+  cancelableMessage
+}: {
+  cancelableMessage: CancelableMessage
+}) {
+  const theme = useTheme()
+  const { i18n } = useTranslation()
+  const { refreshMessages, setCancelableMessage, selectThread } =
+    useContext(MessageContext)
+  const [secondsLeft, setSecondsLeft] = useState<number>(
+    getSecondsLeft(cancelableMessage.sentAt)
```

Comment:
Nitpick: Use `() => getSecondsLeft(...)` to avoid useless computing of the initial value on each render.